BLSQ/iaso
# IA-1301: 401 unauthorize

When a project requires authentication (feature flag) or an endpoint does, if no authentication token is provided (or the authentication token is expired), the backend should return a 401 - Unauthorized to indicate the endpoint requires a valid authentication token to be provided.

Related JIRA tickets :IA-1301, WC2-353

## Self proofreading checklist


## Doc
- https://www.django-rest-framework.org/api-guide/authentication/#unauthorized-and-forbidden-responses
- https://auth0.com/blog/forbidden-unauthorized-http-status-codes/

## Changes

- just reorder AUTH_CLASSES with `CsrfExemptSessionAuthentication` at the end to return 401
- Tweak `iasoFetch` to redirect to login if any call receive a 401
- fix python tests 
- fix logout redirect url 

## How to test

Use PostMan to GET a secure API endpoint like `/api/payments/lots/` without auth token
You should receive a 401 not a 403

Open the dashboard, login on one tab and open a second one on the dashboard too.
Logout on one of those tab and try to make something on the second, you should be redirect to login page

## Print screen / video

![Screenshot 2024-03-20 at 11 27 53](https://github.com/BLSQ/iaso/assets/12494624/4aa54187-b859-4470-8c82-14743f0bb45d)
